File: Pet-Adoption-Shelter-Front/src/apis/adoptions/adoption.api.ts
import type {
  CreateAdoptionReq,
  UpdateAdoptionReq,
  AdoptionRes,
} from "@/types/adoptions/adoption.dto";

import type { ApiResponse } from "@/types/common/ApiResponse";
import { privateApi } from "../common/axiosInstance";
import { ADOPTION_PATH } from "./adoption.path";

export const adoptionsApi = {
  // 입양 계약 생성
  create: async (req: CreateAdoptionReq): Promise<AdoptionRes> => {
    const res = await privateApi.post<ApiResponse<AdoptionRes>>(
      ADOPTION_PATH.ROOT,
      req
    );
    return res.data.data;
  },

  // 입양 계약 상세 조회
  getById: async (adoptionId: number): Promise<AdoptionRes> => {
    const res = await privateApi.get<ApiResponse<AdoptionRes>>(
      ADOPTION_PATH.BY_ID(adoptionId)
    );
    return res.data.data;
  },

  // 입양 계약 수정
  update: async (
    adoptionId: number,
    req: UpdateAdoptionReq
  ): Promise<AdoptionRes> => {
    const res = await privateApi.put<ApiResponse<AdoptionRes>>(
      ADOPTION_PATH.BY_ID(adoptionId),
      req
    );
    return res.data.data;
  },
};

File: mini-proj-Pet-Adoption-Shelter/src/main/java/org/example/miniprojpetadoptionshelter/common/apis/adoption/AdoptionApi.java
package org.example.miniprojpetadoptionshelter.common.apis.adoption;

import org.example.miniprojpetadoptionshelter.common.apis.ApiBase;

public class AdoptionApi {

    private AdoptionApi() {}

    // 최종 엔드포인트: /api/v1/adoptions
    public static final String ROOT = ApiBase.BASE + "/adoptions";

    // 단건 조회/수정: /api/v1/adoptions/{adoptionId}
    public static final String BY_ID = "/{adoptionId}";

    // UPDATE도 동일 경로 사용
    public static final String UPDATE = "/{adoptionId}";
}